packages/runtime/lib/runtime.js

```diff
@@ -1720,62 +1729,70 @@ class Runtime extends EventEmitter {
 
     worker.stdout.setEncoding('utf8')
     worker.stdout.on('data', raw => {
-      let selector = selectors.stdout
-
       if (raw.includes(kStderrMarker)) {
-        selector = selectors.stderr
-        raw = raw.replaceAll(kStderrMarker, '')
+        this.#forwardThreadLog(logger, selectors.stderr, raw.replaceAll(kStderrMarker, ''), 'stderr')
+      } else {
+        this.#forwardThreadLog(logger, selectors.stdout, raw, 'stdout')
       }
-
-      this.#forwardThreadLog(logger, selector, raw)
     })
 
     // Whatever is outputted here, it come from a direct process.stderr.write in the thread.
     // There's nothing we can do about it in regard of out of order logs due to a Node bug.
     worker.stderr.setEncoding('utf8')
     worker.stderr.on('data', raw => {
-      this.#forwardThreadLog(logger, selectors.stderr, raw)
+      this.#forwardThreadLog(logger, selectors.stderr, raw, 'stderr')
     })
   }
 
-  #forwardThreadLog (logger, { level, caller }, data) {
-    if (!this.#loggerDestination) {
+  // label is the key in the logger object, either 'stdout' or 'stderr'
+  #forwardThreadLog (logger, { level, caller }, data, label) {
+    // When captureStdio is false, write directly to the logger destination
+    if (!this.#configManager.current.logger.captureStdio) {
+      this.#stdio[label].write(data)
       return
     }
 
     let plainMessages = ''
     for (const raw of data.split('\n')) {
       // First of all, try to parse the message as JSON
       let message
+      let json
       // The message is a JSON object if it has at least 2 bytes
       if (raw.length >= 2) {
         try {
           message = JSON.parse(raw)
-        } catch (e) {
+          json = true
+        } catch {
           // No-op, we assume the message is raw
         }
       }
 
-      // Not a Pino JSON, accumulate the message and continue
-      if (typeof message?.level !== 'number' || (typeof message?.time !== 'number' && typeof message?.msg !== 'string')) {
-        plainMessages += (plainMessages.length ? '\n' : '') + raw
+      const pinoLog = typeof message?.level === 'number' && typeof message?.time === 'number' && typeof message?.msg === 'string'
```

Comment:
How would this work with customLevels?

Reply:
This works only with default pino settings
Using any of `messageKey`, `formatter.level`, `timestamp` or `customLevel` causes the logger entry to not be recognized as a pino entry and so threated as a json log entry

I think I should I add that to the doc

Reply:
please do